compiler/src/dmd/dinterpret.d

```diff
@@ -6368,6 +6436,29 @@ public:
     {
         assert(0); // This should never be interpreted
     }
+}
+
+/// Interpret `throw <exp>` found at the specified location `loc`
+void interpretThrow(ref Expression result, Expression exp, const ref Loc loc, InterState* istate)
```

Comment:
Why is this function moved and made public?

Reply:
It was moved because it is called from both the Statement code and the Expression code, so it cannot be a nested function of either. It could be private, though.

Reply:
I made no attempt to fix the comments, etc., to keep this an easy cut&paste job as much as practical.